Convert Travis tests to GitHub actions

cesarizu opened the pull request:
This PR converts Travis tests to GitHub actions. Currently all tests that pass in Travis are passing here as well. There are new tests using MongoDB 4.2.

Currently includes changes from: https://github.com/mongoid/mongoid-locker/pull/92

cesarizu commented:
On the logs I see `Could not find the type of CI for Danger to run on.`. Seems that `danger` is outdated and newer versions support GitHub Actions. But `mongoid-danger` (from where this dependency comes from) hasn't been updated in 4 years.

dblock commented:
Let’s leave a Travis config that runs Danger for now?

dblock commented:
Can we put back a travis.yml that runs danger? We do want that for PR linting.

dblock commented on spec/support/locker_is_included_shared_examples.rb:
```diff
@@ -128,7 +128,7 @@
       expect do
         criteria = model.where(_id: 1).unlocked.where(_id: 2)
-        expect(criteria.selector['_id']).to eq(2)
+        expect(criteria.selector['_id']).to eq(Mongoid::VERSION < '7' ? 2 : 1)
```
You can't compare versions like that. It will break at Mongoid 10 for example.

Use https://github.com/dblock/mongoid-compatibility or roll your own, see https://code.dblock.org/2020/07/16/comparing-version-numbers-in-ruby.html

cesarizu replied:
Done, updated code to use `mongoid-compatibility`

cesarizu commented:
I added back a `travis.yml` that runs danger and updated the PR to include the fixes for mongoid 7 and the fix for mongodb 4 from https://github.com/mongoid/mongoid-locker/pull/92.

dblock commented:
I merged this as is.
